includes/wikia/GlobalFunctions.php

```diff
@@ -1562,14 +1562,13 @@ function wfHttpsToHttp( $url ) {
 }
 
 function wfHttpsAllowedForURL( $url ): bool {
-	global $wgWikiaBaseDomain, $wgDevDomain, $wgWikiaEnvironment;
-	$parsedURL = parse_url( $url );
-	if ( $parsedURL === false ) {
+	global $wgWikiaBaseDomain, $wgDevDomain, $wgWikiaEnvironment, $wgDevelEnvironment;
+	$host = parse_url( $url, PHP_URL_HOST );
+	if ( $host === false ) {
 		return false;
 	}
 
-	$host = $parsedURL['host'];
-	if ( !empty( $wgDevDomain ) ) {
+	if ( $wgDevelEnvironment && !empty( $wgDevDomain ) ) {
```

Comment:
Do we need to check both?

Reply:
Probably don't but I'm not 100% sure that those two are set always

Reply:
Both are set at the same time: https://github.com/Wikia/config/blob/dev/LocalSettings.php#L19-L21 but doesn't hurt I suppose. :)

Reply:
Especially on tests when they are being mocked out